### PR TITLE
Read Value from Dynamic Sample Store

### PR DESCRIPTION
**Pull request by sonndinh**

### CODE REVIEW

File: src/dds_data.cpp [hunk 1]
```diff
@@ -209,8 +207,223 @@ QVariant CommonData::readValue(const QString& topicName,
...
+    DDS::ReturnCode_t rc = DDS::RETCODE_OK;
+
+    switch (member_tk) {
+    case OpenDDS::XTypes::TK_BOOLEAN:
+      {
+          CORBA::Boolean tmp;
+          rc = parent_data->get_boolean_value(tmp, id);
+          if (rc == DDS::RETCODE_OK) {
+              value = tmp;
+              failed = false;
```
Comment (mitza-oci) [inline src/dds_data.cpp:268]: Would it be simpler and clearer to just return the value here?

File: src/dds_data.cpp [hunk 2]
```diff
@@ -209,8 +207,223 @@ QVariant CommonData::readValue(const QString& topicName,
...
+}
 
-} // End CommonData::readValue
+//------------------------------------------------------------------------------
+QVariant CommonData::readValue(const QString& topicName,
+                               const QString& memberName,
+                               unsigned int index)
+{
+    const QVariant value = readMember(topicName, memberName, index);
+    if (value.toString() != "NULL") {
```
Comment (mitza-oci) [inline src/dds_data.cpp:422]: Would `value.isValid()` or `isNull()` work here?

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: src/dds_data.cpp (hunk 1 above)
Comment (sonndinh) [reply src/dds_data.cpp:268]: Yes, I think so.

File: src/dds_data.cpp (hunk 2 above)
Comment (sonndinh) [reply src/dds_data.cpp:403]: That's true. It looks like a separate problem. Should we fix that in this PR or are you okay with a new PR for that?

File: src/dds_data.cpp [hunk 3]
```diff
@@ -213,45 +213,56 @@ QVariant CommonData::readDynamicMember(const QString& topicName,
...
-        value = "NULL";
-        return value;
+    if (static_cast<int>(index) >= sampleList.count()) {
+        return error;
     }
 
     DDS::DynamicData_var sample = sampleList.at(index);
     DDS::DynamicType_var topic_type = sample->type();
+    OpenDDS::XTypes::MemberPath member_path;
+    if (member_path.resolve_string_path(topic_type, memberName.toStdString().c_str()) != DDS::RETCODE_OK) {
```
Comment (iguessthislldo) [inline src/dds_data.cpp:231]:
```suggestion
    if (member_path.resolve_string_path(topic_type, memberName.toStdString()) != DDS::RETCODE_OK) {
```

File: src/dds_data.cpp (hunk 2 above)
Comment (mitza-oci) [reply src/dds_data.cpp:403]: Consider this potential change in this PR: `readDynamicMember()` returns a null variant on error, but the UI will never see that.  Instead it's used internally to switch to `readMember()` to get a value for the UI.

File: src/dds_data.cpp (hunk 3 above)
Comment (sonndinh) [reply src/dds_data.cpp:231]: This assumes we're not using `OPENDDS_ALLOCATOR`. I think that assumption is safe for the monitor, is that correct?

File: src/dds_data.cpp (hunk 2 above)
Comment (sonndinh) [reply src/dds_data.cpp]: That sounds good. I've updated.

**Review by mitza-oci — APPROVED** [review]

File: src/dds_data.cpp (hunk 3 above)
Comment (mitza-oci) [reply src/dds_data.cpp]: Yes, I don't think the monitor could (or should) be built using a safety profile build of OpenDDS.

Comment (sonndinh) [reply src/dds_data.cpp]: Yeah, that's right. The safety profile build doesn't have dynamic data.

**Pull request merged**